### PR TITLE
Add motivation for real entities.

### PR DESCRIPTION
**Pull request by stefanoborini**

### CODE REVIEW

**Comment by stefanoborini** [comment]
Note: also moved is_binary_vector under the integers, as it fits there.

**Review by dkarletsos — APPROVED** [review]

File: R/integers.R [hunk 1]
```diff
@@ -477,35 +477,51 @@ is_binary_vector <- function(
...
+  res <- inspect_binary_vector(
+    v, allow_na_values = allow_na_values, allow_degenerate = allow_degenerate
+  )
+
+  return(res$valid)
+}
+assertthat::on_failure(is_binary_vector) <- function(call, env) {
+  v <- callget(call, env, "v", NULL)
+  allow_na_values <- callget(call, env, "allow_na_values", FALSE)
+  allow_degenerate <- callget(call, env, "allow_degenerate", TRUE)
```
[the remark this answers is not in the document]
Comment (stefanoborini) [reply R/integers.R:489]: To be fair, I really don't like the "degenerate" option in is_binary_vector. I discussed at length with Boryana on how to call it, and we could not find a better name, but you are right. It's kind of deceiving that they are being used in different contexts, and I didn't realise it before. Do you have a better name for it?

[the remark this answers is not in the document]
Comment (stefanoborini) [reply R/integers.R:489]: maybe allow_uniform?

[the remark this answers is not in the document]
Comment (dkarletsos) [reply R/integers.R:489]: agreed

[the remark this answers is not in the document]
Comment (stefanoborini) [reply R/integers.R:489]: Decided to change the parameter name in a separate PR.

**Pull request merged**